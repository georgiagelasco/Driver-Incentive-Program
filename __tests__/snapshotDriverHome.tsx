/**
 * @jest-environment jsdom
 */

import { render, waitFor } from "@testing-library/react";
import Page from "../app/driver/home/page";
import React from 'react'

jest.mock('next/navigation', () => ({
    useRouter: jest.fn(() => ({
      push: jest.fn(),
      replace: jest.fn(),
      prefetch: jest.fn(),
      back: jest.fn(),
      forward: jest.fn(),
      pathname: '/',
      query: {},
    })),
  }));

  jest.mock("aws-amplify/auth", () => ({
    fetchUserAttributes: jest.fn(() => Promise.resolve({
      "custom:role": "driver",
    })),
    getCurrentUser: jest.fn(() => Promise.resolve({
        username: "test-user",
        attributes: {
          email: "test@example.com",
        },
      })),
  }));

it("renders driver homepage unchanged", async () => {
  const { container } = render(<Page />);

    await waitFor(() => {
      expect(container).toBeTruthy(); // This triggers a wait for React updates
    });

  expect(container).toMatchSnapshot();
});