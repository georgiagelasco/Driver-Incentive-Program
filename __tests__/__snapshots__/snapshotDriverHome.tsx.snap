// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders driver homepage unchanged 1`] = `
<div>
  <div
    class="flex flex-col h-screen"
  >
    <nav
      class="flex justify-between items-center bg-gray-800 p-4 text-white"
    >
      <div
        class="flex gap-4"
      >
        <button
          class="bg-blue-600 px-4 py-2 rounded text-white"
        >
          Home
        </button>
        <a
          href="/aboutpage"
        >
          <button
            class="bg-gray-700 px-4 py-2 rounded hover:bg-gray-600"
          >
            About Page
          </button>
        </a>
        <a
          href="/driver/driver_cat"
        >
          <button
            class="bg-gray-700 px-4 py-2 rounded hover:bg-gray-600"
          >
            Catalog
          </button>
        </a>
        <a
          href="/driver/points"
        >
          <button
            class="bg-gray-700 px-4 py-2 rounded hover:bg-gray-600"
          >
            Points
          </button>
        </a>
        <a
          href="/driver/driver_app"
        >
          <button
            class="bg-gray-700 px-4 py-2 rounded hover:bg-gray-600"
          >
            Application
          </button>
        </a>
      </div>
      <div
        class="relative"
      >
        <div
          class="cursor-pointer text-2xl"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
        </div>
      </div>
    </nav>
    <main
      class="flex-grow flex flex-col items-center justify-center p-10"
    >
      <h1
        class="text-5xl font-light mb-4 text-center"
      >
        Welcome, 
        No email found
      </h1>
      <p
        class="text-lg text-center mb-8"
      >
        You are logged in as a driver, a true Mother Trucker! You can view your sponsor(s) below. Don't have a sponsor yet? Click the 'Application' tab to fill out your application.
      </p>
      <p
        class="text-center text-lg"
      >
        You are not currently connected to a sponsor company. Please click the “Application” tab above to apply.
      </p>
    </main>
  </div>
</div>
`;
